x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list.tsx

```diff
@@ -47,11 +47,22 @@ const CreateButton: React.FC = () => {
 };
 
 export const EnginesList: React.FC = () => {
-  const { closeEngineCreate, fetchEngines, onPaginate, openDeleteEngineModal, setSearchQuery } =
-    useActions(EnginesListLogic);
+  const {
+    closeEngineCreate,
+    fetchEngines,
+    onPaginate,
+    openDeleteEngineModal,
+    setIsFirstRequest,
+    setSearchQuery,
+  } = useActions(EnginesListLogic);
   const { openFetchEngineFlyout } = useActions(EnginesListFlyoutLogic);
-  const { isLoading, meta, results, createEngineFlyoutOpen, searchQuery } =
+  const { createEngineFlyoutOpen, hasNoEngines, isLoading, meta, results, searchQuery } =
     useValues(EnginesListLogic);
+  useEffect(() => {
+    // We don't want to trigger loading for each search query change, so we need this
+    // flag to set if the call to backend is first request.
+    setIsFirstRequest();
```

Comment:
Nice! I guess this would also solve existing page flickering [issue](https://github.com/elastic/enterprise-search-team/issues/3772)[issue](https://github.com/elastic/enterprise-search-team/issues/3772) in list page UI